modules/planning/scenarios/stage.cc

```diff
@@ -61,11 +68,11 @@ Stage::Stage(const ScenarioConfig::StageConfig& config) : config_(config) {
     CHECK(config_map.find(task_type) != config_map.end())
         << "Task: " << TaskConfig::TaskType_Name(task_type)
         << " used but not configured";
-    auto iter = tasks_.find(task_type);
-    if (iter == tasks_.end()) {
+    auto iter = tasks_[cur_stage_].find(task_type);
+    if (iter == tasks_[cur_stage_].end()) {
       auto ptr = TaskFactory::CreateTask(*config_map[task_type]);
       task_list_.push_back(ptr.get());
-      tasks_[task_type] = std::move(ptr);
+      tasks_[cur_stage_][task_type] = std::move(ptr);
```

Comment:
When you use a "static" task_map, a stage type will be put into this map and stay.  
line 73 - 75 will not be executed.  A second run of "the same stage_type" may not have the same config file, and therefore this stage won't be constructed properpy with the corresponding conf file. 

For instance, CREEP stage in scenario A may have different set of conf from CREEP stage in scenario B.